src/cargo/core/registry.rs

```diff
@@ -174,14 +177,18 @@ impl<'cfg> PackageRegistry<'cfg> {
         sub_vec.push((id, deps));
     }
 
-    fn load(&mut self, source_id: &SourceId, kind: Kind) -> CargoResult<()> {
+    fn load(&mut self, source_id: &SourceId, kind: Kind, 
+            updated: &mut HashSet<String>) -> CargoResult<()> {
```

Comment:
How come this parameter was added? There's other machinery in play here to ensure that we don't update sources twice, so I'm curious what triggered this path.